app/helpers/files.js

```diff
@@ -0,0 +1,58 @@
+import fs from "fs-extra";
+
+// readFileBackward reads a file backward and if maxSize is specified it will
+// only read until reach that size in bytes.
+export function readFileBackward(path, maxSize, end) {
+  fs.open(path, "r", function (error, descriptor) {
+    if (error) return end(error);
+    fs.fstat(descriptor, function (error, stats) {
+      if (error) return end(error);
+      let buffer, position;
+      if (stats.size > maxSize) {
+        buffer = Buffer.alloc(maxSize);
+        position = stats.size - maxSize;
+      } else {
+        buffer = Buffer.alloc(stats.size);
+        position = 0;
+      }
+      let offset = 0;
+      let length = buffer.length;
+      const read = function () {
+        fs.read(descriptor, buffer, offset, length, position, function (
+          error,
+          copied
+        ) {
+          if (error) return end(error);
+          offset += copied;
+          length -= copied;
+          if (length === 0) return end(undefined, buffer);
+          read();
+        });
+      };
+      read();
+    });
+  });
+}
+
+// makeFileBackup makes a backup of the file on the directory specified.
+// If the directory does not exists, it will be created.
+export function makeFileBackup(file, directory) {
+  if (!fs.existsSync(file)) {
+    throw "File does not exists";
+  }
+  // get file name. Which probably is everything after the last /
+  // ex: /home/.dcrd/rpc.cert
+  const fileName = file.substr(file.lastIndexOf("/") + 1);
```

Comment:
Can this break under windows or is `file` guaranteed to use `/` as separator? Usually the correct function to use in these situations is `basename()` (see [path.basename](https://nodejs.org/api/path.html#path_path_basename_path_ext)) but there may be considerations for not using that.

Reply:
Yeah, you are right, that will be a problem. What considerations are that?

Reply:
For example, if `file` is guaranteed to be coming from a hardcoded constant or was previously converted to use slashes independently of the platform, then `basename()` wouldn't work because it would determine the basename by the last `\` in windows.